### PR TITLE
Earn: Convert jsdoc to TypeScript types

### PR DESCRIPTION
**Pull request by edanzer**
## Proposed Changes

Converts jsdoc types to TypeScript types/syntax. This resolves some minor, non-blocking eslint notices about the jsdocs types. It also allows us to tighten up some types - in Earn > Main, we had been returning 'object' types. I've updated that to be the more precise PromoSectionCardProps types. 

All of this represents a small follow up after refactoring all the Earn code to TypeScript over the last month. 

## Testing Instructions

1) Checkout this branch, go to http://calypso.localhost:3000/earn/YOURDOMAIN, and click around to confirm everything works as expected. 

2) In your code editor, review the earn code (especially the changed files) and confirm there are no TypeScript errors. 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/earn/home.tsx [hunk 1]
```diff
@@ -328,7 +323,7 @@ const Home = () => {
 				'https://wordpress.com/support/wordpress-editor/blocks/premium-content-block/'
 			),
 			onClick: () => trackLearnLink( 'premium-content' ),
-			label: hasConnectedAccount ? translate( 'Support documentation' ) : null,
+			label: hasConnectedAccount ? translate( 'Support documentation' ) : undefined,
```
Comment (edanzer) [inline client/my-sites/earn/home.tsx:326]: This needs to be undefined, not null, based on PromoSectionCardProps.

File: client/my-sites/earn/components/add-edit-plan-modal/index.tsx [hunk 2]
```diff
@@ -47,11 +47,6 @@ type DefaultNames = {
  * Return the minimum transaction amount for a currency.
  * If the defaultCurrency is not the same as the current currency, return the double, in order to prevent issues with Stripe minimum amounts
  * See https://wp.me/p81Rsd-1hN
- *
- * @param {StripeMinimumCurrencyAmounts} currency_min - Minimum transaction amount for each currency.
- * @param {string} currency - Currency.
- * @param {string} connectedAccountDefaultCurrency - Default currency of the current account
- * @returns {number} Minimum transaction amount for given currency.
```
Comment (edanzer) [inline client/my-sites/earn/components/add-edit-plan-modal/index.tsx:54]: The types above were redundant give they're defined in the TS notation just below.

File: client/my-sites/earn/home.tsx (hunk 1 above)
Comment (simison) [reply client/my-sites/earn/home.tsx:326]: Weird that we set label undefined, instead of just not rendering the whole thing?

**Review by simison — COMMENTED** [review]
👍 ✅

**Pull request merged**